snuba/util.py

```diff
@@ -102,64 +167,34 @@ def column_expr(column_name, body, alias=None, aggregate=None):
         expr = escape_col(column_name)
 
     if aggregate:
-        if expr:
-            expr = u'{}({})'.format(aggregate, expr)
-            if aggregate == 'uniq':  # default uniq() result to 0, not null
-                expr = 'ifNull({}, 0)'.format(expr)
-        else:  # This is the "count()" case where the '()' is already provided
-            expr = aggregate
+        expr = function_expr(aggregate, expr)
 
     alias = escape_col(alias or column_name)
 
     return alias_expr(expr, alias, body)
 
 
 def complex_column_expr(expr, body, depth=0):
-    # TODO instead of the mutual recursion between column_expr and complex_column_expr
-    # we should probably encapsulate all this logic in a single recursive column_expr
-    if depth == 0:
-        # we know the first item is a function
-        ret = expr[0]
-        assert SAFE_FUNCTION_RE.match(ret)
-        expr = expr[1:]
-
-        # if the last item of the toplevel is a string, it's an alias
-        alias = None
-        if len(expr) > 1 and isinstance(expr[-1], six.string_types):
-            alias = expr[-1]
-            expr = expr[:-1]
-    else:
-        # is this a nested function call?
-        if len(expr) > 1 and isinstance(expr[1], tuple):
-            ret = expr[0]
-            assert SAFE_FUNCTION_RE.match(ret)
-            expr = expr[1:]
+    name, args, alias = is_function(expr, depth)
```

Comment:
I guess we already checked `if is_function(...)` before calling `complex_column_expr`, so we know that this will return a 3-tuple and not `None`, but I'm wondering if we should do:
```
function_tuple = is_function(expr, depth)
if function_tuple is not None:
    name, args, alias = function_tuple
```
just to be more safe in case this starts getting used by other callers.